src/context/AuthContext.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  AuthError,
+  AuthResponse,
+  Session,
+  User,
+  UserAttributes,
+  UserResponse,
+} from '@supabase/supabase-js';
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import supabaseAdmin from '../supabase/createAdminClient';
+import supabase from '../supabase/createClient';
+
+export interface AuthState {
+  session: Session | null;
+  user: User | null;
+  isLoading: boolean;
+  signIn: (newSession: Session | null) => void;
+  signUp: (
+    email: string,
+    password: string,
+    options: object,
+  ) => Promise<AuthResponse>;
+  signInWithEmail: (email: string, password: string) => Promise<AuthResponse>;
```

Comment:
Out of curiosity, if the auth context only wraps the `(BottomTabNavigation)` folder, how can the login flow access the `signInWithEmail()` function, for example?